### PR TITLE
uartEnd: Unlock mutex before detaching rx and tx

### PR DESCRIPTION
**Pull request by Raienryu97**
This should solve the device freezing issue when Serial.end() is called

### CODE REVIEW

**Comment by Raienryu97** [comment]
Verified with the following code:

```
void setup(){
  Serial.begin(9600);
  delay(2000);
  Serial.println("Hello");
  Serial.begin(115200);
}

void loop(){
  Serial.println("Hello Again");
  delay(1000);
}
```

Serial Output
```
ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
ets Jun  8 2016 00:22:57

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0010,len:4
load:0x3fff0014,len:708
load:0x40078000,len:0
load:0x40078000,len:12168
entry 0x40078a68
 !⸮⸮⸮⸮⸮Again
Hello Again
Hello Again
Hello Again
Hello Again
Hello Again
Hello Again
Hello Again
Hello Again
```

**Comment by everslick** [comment]
Ahh seems to be the issue with #513 as well...

**Comment by everslick** [comment]
This indeed fixes my problem with Serial.end(), so im closing #513 ...

File: cores/esp32/esp32-hal-uart.c [hunk 1]
```diff
@@ -227,12 +227,12 @@ void uartEnd(uart_t* uart)
         while(xQueueReceive(uart->queue, &c, 0));
         vQueueDelete(uart->queue);
     }
+    UART_MUTEX_UNLOCK();
 
     uartDetachRx(uart);
     uartDetachTx(uart);
 
     uart->dev->conf0.val = 0;
```
Comment (me-no-dev) [inline cores/esp32/esp32-hal-uart.c]: leave this line inside the lock ;)

**Comment by Raienryu97** [comment]
Done @me-no-dev :)

**Pull request merged**